stubs/babel/babel/numbers.pyi

```diff
@@ -1,99 +1,124 @@
-from typing import Any
+import decimal
+from datetime import date, datetime
+from re import Pattern
+from typing_extensions import Literal
 
 from babel.core import Locale
 
 long = int
-LC_NUMERIC: Any
+LC_NUMERIC: str | None
 
 class UnknownCurrencyError(Exception):
-    identifier: Any
-    def __init__(self, identifier) -> None: ...
+    identifier: str
+    def __init__(self, identifier: str) -> None: ...
 
-def list_currencies(locale: Any | None = ...): ...
-def validate_currency(currency, locale: Any | None = ...) -> None: ...
-def is_currency(currency, locale: Any | None = ...): ...
-def normalize_currency(currency, locale: Any | None = ...): ...
-def get_currency_name(currency, count: Any | None = ..., locale=...): ...
-def get_currency_symbol(currency, locale=...): ...
-def get_currency_precision(currency): ...
-def get_currency_unit_pattern(currency, count: Any | None = ..., locale=...): ...
+def list_currencies(locale: Locale | str | None = ...) -> set[str]: ...
+def validate_currency(currency: str, locale: Locale | str | None = ...) -> None: ...
+def is_currency(currency: str, locale: Locale | str | None = ...) -> bool: ...
+def normalize_currency(currency: str, locale: Locale | str | None = ...) -> str | None: ...
+def get_currency_name(currency: str, count: float | decimal.Decimal | None = ..., locale: Locale | str | None = ...) -> str: ...
+def get_currency_symbol(currency: str, locale: Locale | str | None = ...) -> str: ...
+def get_currency_precision(currency: str) -> int: ...
+def get_currency_unit_pattern(currency: str, count: float | None = ..., locale: Locale | str | None = ...) -> str: ...
```

Comment:
```suggestion
def get_currency_unit_pattern(currency: str, count: float | None = ..., locale: Locale | str = ...) -> str: ...
```
Same here, it does `locale.parse` and blindly thinks the locale will be non-None.